core/src/main/scala/com/databricks/labs/remorph/parsers/intermediate/extensions.scala

```diff
@@ -2,4 +2,18 @@ package com.databricks.labs.remorph.parsers.intermediate
 
 trait AstExtension
 
-case class Column(name: String) extends Expression with AstExtension {}
\ No newline at end of file
+case class Column(name: String) extends Expression with AstExtension {}
+
+abstract class Predicate extends Expression with AstExtension {}
```

Comment:
```suggestion
abstract class Predicate extends Expression with AstExtension {}
abstract class Binary(left: Expression, right: Expression) extends Predicate {}
abstract class Unary(left: Expression) extends Predicate {}
```
having this separation would help